lib/SILOptimizer/Analysis/BasicCalleeAnalysis.cpp

```diff
@@ -142,12 +142,28 @@ void CalleeCache::computeWitnessMethodCalleesForWitnessTable(
       continue;
     }
 
-    auto Witness = WT.getConformance()->getWitness(Requirement.getDecl(),
-                                                   nullptr);
-    auto DeclRef = SILDeclRef(Witness.getDecl());
-
-    bool canCallUnknown = !calleesAreStaticallyKnowable(M, DeclRef);
-
+    bool canCallUnknown = false;
+
+    auto Conf = WT.getConformance();
+    switch (Conf->getProtocol()->getEffectiveAccess()) {
```

Comment:
Checking only the effective access of the protocol is safe, but overly conservative: I think you want the minimum effective access of the conforming type and the protocol, because that tells you how much knowledge of you have of the conformance itself. I would be reasonable to put that operation onto `ProtocolConformance`.

Reply:
Apparently, I misunderstood what this analysis is doing, so this is not overly conservative.